scalableminds/webknossos
# Exponential backoff for retrying saving

The save retry time will increase exponentially, but will be `MAX_SAVE_RETRY_WAITING_TIME` at most, which I've set to 5 Minutes for now (any thoughts on that value?).

![exp-backoff](https://user-images.githubusercontent.com/1702075/52275087-d8067300-294e-11e9-94b6-f01b5262f9f5.png)

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Go offline - (Network Tab - Offline)
- Set a node and press save, monitor the save requests in the network tab, they should be after 2s, 4s, 8s, and so on, up until 5min.
- Go back online, saving should succeed eventually.
OR
- Pressing the save button (or Ctrl + S) should immediately save the tracing.

### Issues:
- fixes #3012 

------
